runtime/src/iree/tooling/parameter_util.c

```diff
@@ -0,0 +1,276 @@
+
+IREE_FLAG(
+    string, parameter_mode, "mmap",
+    "A parameter I/O mode of ['preload', 'mmap'].\n"
+    "  preload: read entire parameter files into wired memory on startup.\n"
+    "  mmap: maps the parameter files into discardable memory - can increase\n"
+    "        warm-up time and variance as mapped pages are swapped\n"
+    "        by the OS.");
+
+static void iree_file_contents_release_callback(
+    void* user_data, iree_io_file_handle_primitive_t handle_primitive) {
+  iree_file_contents_t* file_contents = (iree_file_contents_t*)user_data;
+  iree_file_contents_free(file_contents);
+}
+
+// Opens the parameter file at |path| with the mode specified by the
+// --parameter_mode flag and returns its handle.
+static iree_status_t iree_io_open_parameter_file(
+    iree_string_view_t path, iree_allocator_t host_allocator,
+    iree_io_file_handle_t** out_file_handle) {
+  IREE_ASSERT_ARGUMENT(out_file_handle);
+  *out_file_handle = NULL;
+  IREE_TRACE_ZONE_BEGIN(z0);
+  IREE_TRACE_ZONE_APPEND_TEXT(z0, path.data, path.size);
+
+  char path_str[2048] = {0};
+  iree_string_view_to_cstring(path, path_str, sizeof(path_str));
+  iree_file_read_flags_t read_flags = 0;
+  if (strcmp(FLAG_parameter_mode, "mmap") == 0) {
+    read_flags |= IREE_FILE_READ_FLAG_MMAP;
+  } else if (strcmp(FLAG_parameter_mode, "preload") == 0) {
+    read_flags |= IREE_FILE_READ_FLAG_PRELOAD;
+  } else {
+    IREE_TRACE_ZONE_END(z0);
+    return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
+                            "unrecognized --parameter_mode= value '%s'",
+                            FLAG_parameter_mode);
+  }
+
+  iree_file_contents_t* file_contents = NULL;
+  IREE_RETURN_AND_END_ZONE_IF_ERROR(
+      z0, iree_file_read_contents(path_str, read_flags, host_allocator,
+                                  &file_contents));
+
+  iree_io_file_handle_release_callback_t release_callback = {
+      .fn = iree_file_contents_release_callback,
+      .user_data = file_contents,
+  };
+  iree_io_file_handle_t* file_handle = NULL;
+  iree_status_t status = iree_io_file_handle_wrap_host_allocation(
+      IREE_IO_FILE_ACCESS_READ, file_contents->buffer, release_callback,
+      host_allocator, &file_handle);
+  if (iree_status_is_ok(status)) {
+    *out_file_handle = file_handle;
+  } else {
+    iree_file_contents_free(file_contents);
+  }
+  IREE_TRACE_ZONE_END(z0);
+  return status;
+}
+
+//===----------------------------------------------------------------------===//
+// Parameter file format parsing
+//===----------------------------------------------------------------------===//
+
+IREE_FLAG_LIST(
+    string, parameters,
+    "Specifies a parameter file to make available to programs with either an\n"
+    "anonymous global scope (`some_file.gguf`) or a named scope like\n"
+    "`my_scope=some_file.gguf`.\n"
+    "\n"
+    "Supported formats:\n"
+    "- .irpa (IREE parameter archive)\n"
+    "- .gguf (https://github.com/philpax/ggml/blob/gguf-spec/docs/gguf.md)"
```

Comment:
update link: https://github.com/ggerganov/ggml/blob/master/docs/gguf.md
```suggestion
    "- .gguf (https://github.com/ggerganov/ggml/blob/master/docs/gguf.md)"
```